ipaserver/install/cainstance.py

```diff
@@ -1649,6 +1686,9 @@ def __store_random_serial_number_state(self):
         dn = DN(('cn', ipalib.constants.IPA_CA_CN), api.env.container_ca,
                 api.env.basedn)
         entry_attrs = api.Backend.ldap2.get_entry(dn)
+        version = entry_attrs.get("ipaCaRandomSerialNumberVersion", "x")
```

Comment:
This line should be ` version = entry_attrs.single_value.get("ipaCaRandomSerialNumberVersion", "x")` otherwise version = ['3'] and str(version) = "['3']" which leads to update_entry with the same value.